### PR TITLE
[TACHYON-597] recache() allocates too much memory

### PR DESCRIPTION
**Pull request by miasantreble**
https://tachyon.atlassian.net/browse/TACHYON-597

### CODE REVIEW

**Comment by calvinjia** [comment]
+1 LGTM

File: clients/unshaded/src/main/java/tachyon/client/TachyonFile.java [hunk 1]
```diff
@@ -495,7 +495,7 @@ boolean recache(int blockIndex) throws IOException {
       inputStream.skip(offset);
 
       int bufferBytes =
-          (int) mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB) * 4;
+          (int) mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB);
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFile.java]: Was the original intention to default to 4MB? Is one buffer size more appropriate than the other?

Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFile.java]: maybe I misread the ticket: "However, the "*4" should probably be with the Constants.MB to be 4 MB"
are you saying the call to getBytes should be something like this?
mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB*4)
making each unit buffer of size 4MB?

Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFile.java]: I think it should be 1 MB given the default property value file. You can open another JIRA ticket to investigate what size is ideal in most situations, but I think it should be part of a separate PR.

**Comment by gpang** [comment]
LGTM
